test/e2e/basic-auth.e2e.test.js

```diff
@@ -73,7 +73,7 @@ describe('E2E: basic-auth Policy', () => {
 
   after((done) => {
     gatewayProcess.kill();
-    backendServer.close(done);
+    backendServers[0].close(done);
```

Comment:
Good point, this is a bit tricky.

`startGatewayInstance` has a parameters asking how many backend servers you want to create with the gateway instance, and by default is `1`; therefore I know ahead of the time that the array is going to have only one element.

In [this other part](https://github.com/ExpressGateway/express-gateway/pull/695/files#diff-303a9d51bb3be7cc500aa5c5d5c0aeebR17) for example, I'm asking 2 backend servers (for load balancer tests) therefore I'm closing two elements.

### Hey, why not loop in the array and close all of them?

Because closing the server is an async operation, so looping is not an option. We would need not a loop, but rather a `reduce` function call that'd kind of close the servers one by one, waiting for the callback, and then moving next.

The result would be a really unreadable code, that's why I went in this direction.

#### Byproduct: async.js

However, [async](https://github.com/caolan/async) would resolve such thing, as it provides an high order series of functions dealing with callbacks by default. In particular, [each](http://caolan.github.io/async/docs.html#each) would solve the problem in a very elegant way. I haven't found a way to include only that function and not the whole library. 

Moreover, I think this helper should be refactored yet another time anyway, since the control flow is a little bit shared (leaky, I would say) between the helper and the caller with no clear boundaries — that's why I really didn't want to spend that much time on make it nice. #579 is the tracking card.

I'm fine with all the solutions; tell me the one you prefer, and I'll go toward it.